### PR TITLE
Improve

### PR DESCRIPTION
**Pull request by KidkArolis**
- use eslint output by default as suggested in #10 
- allow configuring output with --format
- preserve standard output with --format standard so editor plugins can continue working as is for now
- remove header and --fix suggestions from output
- add prettier rules for all frameworks as suggested in #9 
- move prettier to dev dependencies as suggested in #8 
- prettier is now entirely optional, if it's there eslint-plugin-prettier will be used and --fix becomes available

Still working on this, but feedback welcome @sheerun.

In particular, if I entirely remove eslint-plugin-prettier, then the healthier output no longer contains prettier style issues, e.g. if you have some semicolons somewhere, I thought it was good that healthier shows that, or else you might not immediately realise that prettier wasn't run on those files or your prettier is misconfigured and always misses some files. Similarly, healthier --fix can be a very convenient shortcut for running prettier on all the files required. This does blur the line a bit, so I'm still not sure about this, but the point was to make this entirely optional. So if prettier is not installed into the parent project (with whatever version is used in the project), then this feature is off, basically we switch to a different eslintrc.json. The only thing I want to double check is whether installing healthier now shows warnings since eslint-plugin-prettier expects prettier as peer dep.

Again, I might still remove prettier plugin altogether. But that would mean losing the helpful linting about prettier style issues. This way, the tool works well with prettier, but prettier is entirely optional now. If it's not there, the style linting is off.

### CODE REVIEW

**Comment by KidkArolis** [comment]
Yeah, it does warn about the peer dep, which is a bit annoying, hm..

```
npm WARN eslint-plugin-prettier@3.0.1 requires a peer of prettier@>= 1.13.0 but none is installed. You must install peer dependencies yourself.
```

**Comment by sheerun** [comment]
> I thought it was good that healthier shows that, or else you might not immediately realise that prettier wasn't run on those files or your prettier is misconfigured and always misses some files.

I think it's better to always skip prettier issues and remove `--fix` all together, this way healthier can focus entirely on linting instead of formatting, and you don't need to discover other tools like prettierx or prettier-standard

As for peer dependencies warnings I'd bundle dependencies instead of listing them as deps

**Comment by sheerun** [comment]
Also, eslint-prettier-plugin is terrible, please don't use it. It passes each linted file though prettier which is very slow.

**Comment by KidkArolis** [comment]
Yeah, as usual, you're convincing me, I will remove --fix and just focus on linting non style issues.

I wouldn't say eslint-plugin-prettier is terrible, it's just a clever way of getting the prettier issues reported. I gueeeees, I should just be using prettier --check in CI instead of using healthier to do the style linting?

The only thing is:

```json
"scripts": {
  "lint": "healthier"
}
```
is a bit easier to use than

```json
"scripts": {
  "lint": "healthier && prettier --check '**/*.js'"
}
```

And healthier is all about convenience.

Here are my current considerations:

* ✅ peer dep warnings (can be fixed)
* ⚠️  potential prettier version mismatch between what you installed, and what healthier's eslint-plugin supports, etc. (maybe can be fixed by detecting version of prettier and only supporting certain versions)
* ✅ preferential prettier treatment (minor point, since we can auto detect if it's used or not, and prettier is the most popular choice anyway)
* ⚠️ performance overhead of double parsing, only an issue when you run the tool over all files, it's fast enough for use within the editor (not sure)
* ⚠️ confusion around tool's scope, does healthier help with code quality, or does it also help with knowing if code is formatted (?)

TBH, `healthier && prettier --check` is not that bad.

Another thing I _could_ do is implement yet another, separate, even more opinionated package, that combines healthier and prettier into one tool if that's what I'm seeking for here.. That would leave healthier nicely reusable by a wider audience. The only difficulty there is taking on code editor plugin maintenance, since you can no longer defer to prettier's extension and .prettierrc would confuse the extensions. Anyway, I'm going off topic and overthinking this ;)

tl;dr - I will remove --fix, keep the tool focused, simpler and fast. See where we go from there.

**Comment by KidkArolis** [comment]
I will also remove `--format standard`, it will be cleaner and better to just update code editors to parse eslint output (current extensions have issues anyway with parsing linter messages about syntax errors).

**Comment by KidkArolis** [comment]
OK, this is pretty much ready now.

It fixes #7, #8, #9, #10 and #12.

Sublime plugin has been updated and works better now.
I just need to update VSCode plugin. But maybe I can do that after releasing this, since pretty much nobody is using this yet.
